apps/src/templates/sectionProgress/sectionProgressRedux.js

```diff
@@ -222,14 +222,35 @@ export default function sectionProgress(state=initialState, action) {
           course.script_ids.forEach(id => idMap[id] = true);
         }
       });
-
       validScripts = validScripts.filter(script => idMap[script.id]);
+
+      var scriptId;
+      switch (true) {
+        // When there is a scriptId already in state.
+        case !!state.scriptId:
+          scriptId = state.scriptId;
+          break;
+        // When there is an assigned course, set scriptId to the first script in the assigned course.
+        case !!action.assignedCourseId:
+          action.validCourses.forEach(course => {
+            if (course.id === action.assignedCourseId) {
+              scriptId = course.script_ids[0];
+            }
+          });
+          break;
+        // If there are validScripts, set scriptId to the first valid script.
+        case validScripts.length > 0:
+          scriptId = validScripts[0].id;
+          break;
+        default:
+         scriptId = ACCELERATED_SCRIPT_ID;
```

Comment:
it should be safe to remove this default case, because validScripts will never be empty (always contains the accelerated course).